torch/_inductor/lowering.py

```diff
@@ -730,7 +730,7 @@ def _to_dtype(x):
     return make_pointwise(_to_dtype, override_return_dtype=dtype)(x)
 
 
-@register_lowering(torch._higher_order_ops._foreach_map)
+@register_lowering(torch._higher_order_ops._foreach_map, type_promotion_kind=None)
```

Comment:
foreach_map handles its own type promotion, otherwise type_promotion_kind = DEFAULT causes it to upcast all of the inputs. This wasn't an option previously because the DEFAULT only upcasts tensors, not tensors in tuples.